charts/prometheus-elasticsearch-exporter/templates/_helpers.tpl

```diff
@@ -0,0 +1,33 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "elasticsearch-exporter.name" -}}
```

Comment:
By best practice, the prefix for the named templates should be the chart name (which was changed during the migration process from `elasticsearch-exporter` to `prometheus-elasticsearch-exporter`)

> One popular naming convention is to prefix each defined template with the name of the chart: {{ define "mychart.labels" }}. By using the specific chart name as a prefix we can avoid any conflicts that may arise due to two different charts that implement templates of the same name.

(https://helm.sh/docs/chart_template_guide/named_templates/)

This would be incompatible with existing releases (since default release name includes the prefx). Any thoughts on this?

Reply:
That's a really good point. I changed the name so that all charts in this repository use prometheus prefix.

I could mention in the docs that one could use `nameOverride` and `fullnameOverride` to keep the old names.

As this chart obviously has a new name I am not sure if we should increment a major version for this or just minor version. What's your view on that?

I also noticed that a container in a pod uses the chart name. That's shouldn't be a problem at all. I personally prefer to have a constant name and not to template that. That makes it easier to get logs from a specific container as I just have to remember that the container is named `exporter` no matter what the Deployment/StatefulSet is called.

Reply:
> As this chart obviously has a new name I am not sure if we should increment a major version for this or just minor version. What's your view on that?

- Since the user needs to adjust configuration (via `nameOverride`/`fullnameOverride`) in order to be able to re-use existing releases with the old naming scheme, i would vote for bumping the major version (incompatible change)
- Adding a migration section in the README would be very handy to have

> I also noticed that a container in a pod uses the chart name. That's shouldn't be a problem at all. I personally prefer to have a constant name and not to template that. That makes it easier to get logs from a specific container as I just have to remember that the container is named exporter no matter what the Deployment/StatefulSet is called.

I agree that container names can be hard coded values.

Reply:
@svenmueller bumped the major version now and added upgrade instructions. Could you please review?

Reply:
Thx! The elasticsearch-exporter chart looks good to me 👍

Reply:
I must be missing something: I still see all templates being prefixed by `elasticsearch-exporter` instead of `prometheus-elasticsearch-exporter`.

Reply:
@desaintmartin I don;t get you here. Where do you see those prefixes?

Most templates use something like `name: {{ template "elasticsearch-exporter.fullname" . }}-cert`

That template is defined like this:

```
{{/*
Create a default fully qualified app name.
We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
If release name contains chart name it will be used as a full name.
*/}}
{{- define "elasticsearch-exporter.fullname" -}}
{{- if .Values.fullnameOverride -}}
{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- $name := default .Chart.Name .Values.nameOverride -}}
{{- if contains $name .Release.Name -}}
{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
{{- end -}}
```

So it should reflect the name of the chart unless you override it to keep the old name. Sure we could rename the template function, but that would not change the rendered content.